mscore/file.cpp

```diff
@@ -1618,6 +1618,8 @@ void MuseScore::exportFile()
       if (lastSaveDirectory.isEmpty())
             lastSaveDirectory = settings.value("lastSaveDirectory", preferences.getString(PREF_APP_PATHS_MYSCORES)).toString();
       QString saveDirectory = lastSaveCopyDirectory;
+      if (cs->masterScore()->fileInfo()->exists())
+            saveDirectory = cs->masterScore()->fileInfo()->dir().path();
```

Comment:
Doesn't this always initialize `saveDirectory` before checking for the current score path? Wouldn't something like

`QString saveDirectory = (cs->masterScore()->fileInfo()->exists() ? cs->masterScore()->fileInfo()->dir().path() : lastSaveCopyDirectory);`

be simpler?

Reply:
Yes it does. Would it be simpler? Probably. The line for "Save As" would become
```
      QString saveDirectory = cs->masterScore()->fileInfo()->exists() ? cs->masterScore()->fileInfo()->dir().path() : saveCopy ? mscore->lastSaveCopyDirectory : mscore->lastSaveDirectory;
```
which is a little long, and maybe a little less clear.

Reply:
I admit I was not considering the efficiency of the QString assignment. If it were simply a pointer or scalar assignment, that would be a different matter.